ui/src/pages/AllStories.tsx

```diff
@@ -0,0 +1,288 @@
+import Paper from "@material-ui/core/Paper";
+import {
+  createStyles,
+  Theme,
+  WithStyles,
+  withStyles,
+} from "@material-ui/core/styles";
+import Switch from "@material-ui/core/Switch";
+import TableCell from "@material-ui/core/TableCell";
+import clsx from "clsx";
+import * as React from "react";
+import {
+  AutoSizer,
+  Column,
+  Table,
+  TableCellRenderer,
+  TableHeaderProps,
+} from "react-virtualized";
+import useSWR from "swr";
+
+import { colors } from "../styles/colors";
+import { Story } from "../types";
+
+declare module "@material-ui/core/styles/withStyles" {
+  interface BaseCSSProperties {
+    flip?: boolean;
+  }
+}
+
+const styles = (theme: Theme) =>
+  createStyles({
+    flexContainer: {
+      display: "flex",
+      alignItems: "left",
+      boxSizing: "border-box",
+    },
+    table: {
+      "& .ReactVirtualized__Table__headerRow": {
+        flip: false,
+        paddingRight: theme.direction === "rtl" ? "0 !important" : undefined,
+        backgroundColor: colors.primaryLight4,
+        fontWeight: 800,
+      },
+      "& .ReactVirtualized__Grid ReactVirtualized__Table__Grid": {},
+    },
+    tableRow: {
+      cursor: "pointer",
+      color: colors.grey,
+    },
+    tableRowHover: {
+      "&:hover": {
+        backgroundColor: colors.black,
+      },
+    },
+    tableCell: {
+      flex: 2,
+    },
+    noClick: {
+      cursor: "initial",
+    },
+  });
+
+interface ColumnData {
+  dataKey: string;
+  label: string;
+  numeric?: boolean;
+  toggle?: true;
```

Comment:
I wanted to do something like `   {columns[columnIndex].toggle && (
          <Switch
            checked={checked}
            onChange={handleChange}
            color="primary"
          />
        )}`
but hooks should not be called conditionally.

Reply:
^ in cell rendering

Reply:
I may be misinterpreting your comment, but I think it's fine to call hooks conditionally as long as they aren't **instantiated** conditionally. @AbhiPrasad pls take a look at this too in case I'm very wrong lmao.

Reply:
Yup according to the rules of hooks you cannot do something like this:

```ts
if (blah) {
  const [state, setter] = React.useState();
}
```

To understand this, you can imagine each useState call being a representing an index in an state array.

So imagine a ReactElement class that represents a React component in the virtual dom.

```
ReactElement {
  state: [state1, state2, state3, ...],
  effects: [...],
  callbacks: [...],
  ref: ...,
  children: [ ... ]
}
```

The first time a component is created, this `state` array is instantiated based on how many times `React.useState()` was called. The `setState` setters from `React.useState()` then rely on the array index to update state. Therefore if you don't call a `React.useState()` because of conditional reasons on a re-render, state/state-setter might reference the wrong index of the state array.

Reply:
Also toggle should just be a boolean, we can handle taking care of it when we grab the data instead of typing the component weirdly.